internal/relationtuple/manager_isolation.go

```diff
@@ -0,0 +1,75 @@
+package relationtuple
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func twice(t *testing.T, m0, m1 Manager) func(string, func(*testing.T, Manager, Manager)) {
+	return func(desc string, run func(*testing.T, Manager, Manager)) {
+		t.Run(desc+"-0", func(t *testing.T) {
+			run(t, m0, m1)
+		})
+		t.Run(desc+"-1", func(t *testing.T) {
+			run(t, m1, m0)
+		})
+	}
+}
+
+func IsolationTest(t *testing.T, m0, m1 Manager, addNamespace func(context.Context, *testing.T, string)) {
+	ctx := context.Background()
+	run := twice(t, m0, m1)
+
+	run("suite=lifecycle", func(t *testing.T, m0, m1 Manager) {
+		nspace := t.Name()
+		addNamespace(ctx, t, nspace)
+
+		rts := []*InternalRelationTuple{
+			{
+				Namespace: nspace,
+				Object:    "a",
+				Relation:  "r",
+				Subject: &SubjectSet{
+					Namespace: nspace,
+					Object:    "o",
+					Relation:  "r",
+				},
+			},
+			{
+				Namespace: nspace,
+				Object:    "b",
+				Relation:  "r",
+				Subject:   &SubjectID{ID: "s"},
+			},
+		}
+
+		t.Run("case=write and get", func(t *testing.T) {
+			require.NoError(t, m0.WriteRelationTuples(ctx, rts...))
+
+			other, _, err := m1.GetRelationTuples(ctx, &RelationQuery{Namespace: nspace})
+			require.NoError(t, err)
+			assert.Len(t, other, 0)
+
+			actual, _, err := m0.GetRelationTuples(ctx, &RelationQuery{Namespace: nspace})
+			require.NoError(t, err)
+			assert.Equal(t, rts, actual)
+		})
+
+		t.Run("case=delete", func(t *testing.T) {
+			require.NoError(t, m1.WriteRelationTuples(ctx, rts...))
+
+			require.NoError(t, m0.DeleteRelationTuples(ctx, rts...))
+
+			deleted, _, err := m0.GetRelationTuples(ctx, &RelationQuery{Namespace: nspace})
+			require.NoError(t, err)
+			assert.Len(t, deleted, 0)
+
+			actual, _, err := m1.GetRelationTuples(ctx, &RelationQuery{Namespace: nspace})
+			require.NoError(t, err)
+			assert.Equal(t, rts, actual)
+		})
```

Comment:
I know that `TransactRelationTuples` uses other internals, but to ensure that all public methods of the persister are tested for isolation, please still add one for `TransactRelationTuples`. Maybe the method changes later on and someone forgets to impelment isolation!